sys/linux/dev_snd_hw.txt

```diff
@@ -0,0 +1,135 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <uapi/asm/ioctl.h>
+include <uapi/linux/fcntl.h>
+include <uapi/linux/time.h>
+include <uapi/sound/asound.h>
+include <uapi/sound/firewire.h>
+
+resource fd_snd_hw[fd]
+
+syz_open_dev$sndhw(dev ptr[in, string["/dev/snd/hwC#D#"]], id intptr, flags flags[open_flags]) fd_snd_hw
+
+read$sndhw(fd fd_snd_hw, buffer ptr[out, snd_firewire_event], count bytesize[buffer])
+read$sndhw_locked(fd fd_snd_hw, buffer ptr[out, snd_firewire_event_lock_status], count bytesize[buffer])
+read$sndhw_fireworks(fd fd_snd_hw, buffer ptr[out, snd_firewire_event_efw_response], count bytesize[buffer])
+read$sndhw_tscm(fd fd_snd_hw, buffer ptr[out, snd_firewire_event_tascam_control], count bytesize[buffer])
+read$sndhw_line6(fd fd_snd_hw, buffer ptr[out, kfifo], count bytesize[buffer])
+
+write$sndhw_fireworks(fd fd_snd_hw, buffer ptr[in, snd_efw_transaction], count bytesize[buffer])
+write$sndhw_line6(fd fd_snd_hw, buffer ptr[in, array[int8]], count bytesize[buffer])
+
+ioctl$SNDRV_HWDEP_IOCTL_PVERSION(fd fd_snd_hw, cmd const[SNDRV_HWDEP_IOCTL_PVERSION], arg ptr[out, int16])
+ioctl$SNDRV_HWDEP_IOCTL_INFO(fd fd_snd_hw, cmd const[SNDRV_HWDEP_IOCTL_INFO], arg ptr[out, snd_hwdep_info])
+ioctl$SNDRV_HWDEP_IOCTL_DSP_STATUS(fd fd_snd_hw, cmd const[SNDRV_HWDEP_IOCTL_DSP_STATUS], arg ptr[out, snd_hwdep_dsp_status])
+ioctl$SNDRV_HWDEP_IOCTL_DSP_LOAD(fd fd_snd_hw, cmd const[SNDRV_HWDEP_IOCTL_DSP_LOAD], arg ptr[in, snd_hwdep_dsp_image])
+ioctl$SNDRV_FIREWIRE_IOCTL_GET_INFO(fd fd_snd_hw, cmd const[SNDRV_FIREWIRE_IOCTL_GET_INFO], arg ptr[out, snd_firewire_get_info])
```

Comment:
Let's keep these ioctls since you've already described them, but they do require real hardware AFAIU. Please add a comment about that above them though:

```
...
ioctl$SNDRV_HWDEP_IOCTL_DSP_LOAD(fd fd_snd_hw, cmd const[SNDRV_HWDEP_IOCTL_DSP_LOAD], arg ptr[in, snd_hwdep_dsp_image])

# These ioctls require actual firewire hardware.
ioctl$SNDRV_FIREWIRE_IOCTL_GET_INFO(fd fd_snd_hw, cmd const[SNDRV_FIREWIRE_IOCTL_GET_INFO], arg ptr[out, snd_firewire_get_info])
...
```

Reply:
And also above `write$sndhw_fireworks` too.

(`write$sndhw_line6` doesn't actually require real hardware, those devices can be emulated via USB, and thus will be tested :)